sdks/python/apache_beam/transforms/trigger.py

```diff
@@ -277,6 +288,10 @@ def on_merge(self, to_be_merged, merge_result, context):
         context.clear_timer('', TimeDomain.WATERMARK)
 
   def should_fire(self, time_domain, watermark, window, context):
+    # Explicitly clear the timer so that late elements are not emitted again
```

Comment:
This comment only applies to the if clause; move there.

Reply:
Done